src/ddmd/dclass.d

```diff
@@ -179,6 +179,26 @@ struct ClassFlags
     alias hasDtor = Enum.hasDtor;
 }
 
+/**
+ * CFM stands for Class Flag Masks. Each ClassDeclaration AST node
+ * allocates a bitfield which stores various class properties.
+ * These properties can be retrieved or updated by using
+ * these masks.
+ */
+enum CFM : ubyte
+{
+    /// mask for COM class (meaning it derives from IUnknown)
+    com = 1,
+    /// mask for a C++ interface
+    cpp = 2,
+    /// mask for an Objective-C class/interface
+    objc = 4,
+    /// mask for a scope class
+    scope_ = 8,
+    /// mask for an anonymous class
+    anon = 16,
+}
```

Comment:
At least some of these are mutual exclusive, if not all? `com`, `cpp`, `objc` and `anon` are at at least mutual exclusive. Then I recommend using `CFM` as the type for `classProperties` and not a mask. Then keep `scope_` as a separate bool instance variable.

Reply:
'Mask' implies they are bits being OR'd together. But exclusive values means that isn't so. I recommend using one field for the exclusive types, and another field for modifiers, instead of mushing them together. I'm guilty of doing the mushing for a long time, and it isn't worth it unless one is desperate to minimize memory usage.
So, I recommend `ClassKind` with members `D`, `com`, `C++`, `anon`. Then, `ClassFlags` with members `stack`. (`scope_` is a bit overloaded in D, and this really is a stack class)

Reply:
**com** and **cpp** aren't exclusive in this case [1] 

[1] https://github.com/dlang/dmd/blob/master/src/ddmd/dclass.d#L976